# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1271109805 14400
#      Mon Apr 12 18:03:25 2010 -0400
# Node ID 62ebba45054e27eb75e967371ccd4db20b65845d
# Parent  ddbd812d09aafdd7839ccebe7d710cd94ad54469
Respect the nextplot property value of 'new' for axes and 'replacechildren' for axes and figures.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,14 @@
+2010-04-12  Ben Abbott <bpabbott@mac.com>
+
+	* plot/__plt_get_axis_arg__.m, plot/newplot.m: Respect the nextplot
+	property value of 'new' for axes and 'replacechildren' for axes and
+	figures.
+
 2010-04-10  Ben Abbott <bpabbot@mac.com>
 
 	* plot/__go_draw_figure__.m: Don't set background for non-visible axes.
 
 2010-04-07  John W. Eaton  <jwe@octave.org>
 
 	* testfun/runtests.m: New function.
 	* testfun/module.mk (testfun_FCN_FILES): Add it to the list.
diff --git a/scripts/plot/__plt_get_axis_arg__.m b/scripts/plot/__plt_get_axis_arg__.m
--- a/scripts/plot/__plt_get_axis_arg__.m
+++ b/scripts/plot/__plt_get_axis_arg__.m
@@ -63,11 +63,15 @@ function [h, varargin, narg] = __plt_get
 	h = gca ();
       endif
     endif
     if (nargin < 2)
       varargin = {};
     endif
   endif
 
+  if (ishandle (h) && strcmp (get (h, "nextplot"), "new"))
+    h = axes ();
+  endif
+
   narg = length (varargin);
 
 endfunction
diff --git a/scripts/plot/newplot.m b/scripts/plot/newplot.m
--- a/scripts/plot/newplot.m
+++ b/scripts/plot/newplot.m
@@ -31,25 +31,28 @@ function newplot ()
     cf = gcf ();
     fnp = get (cf, "nextplot");
     switch (fnp)
       ## FIXME -- probably we should do more than validate the nextplot
       ## property value...
       case "new"
       case "add"
       case "replacechildren"
+        delete (get (cf, "children"))
       case "replace"
       otherwise
 	error ("newplot: unrecognized nextplot property for current figure");
     endswitch
     ca = gca ();
     anp = get (ca, "nextplot");
     switch (anp)
+      case "new"
       case "add"
       case "replacechildren"
+        delete (get (ca, "children"))
       case "replace"
 	__go_axes_init__ (ca, "replace");
 	__request_drawnow__ ();
       otherwise
 	error ("newplot: unrecognized nextplot property for current axes");
     endswitch
   else
     print_usage ();
